### PR TITLE
CB-9692 getAllRecipes throws NPE if recipes field is empty in request object

### PR DESCRIPTION
**Pull request by sodre90**

### CODE REVIEW

**Review by horadla23 — APPROVED** [review]

**Review by keyki — CHANGES_REQUESTED** [review]
Can you please add some unit tests?

**Comment by sodre90** [comment]
sure, I created a unit test for this scenario

File: core-api/src/test/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXV1RequestTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class DistroXV1RequestTest {
+
+    @Test
+    public void getAllRecipesIfRecipeNamesAreNullTest() {
+        DistroXV1Request distroXV1Request = new DistroXV1Request();
+        InstanceGroupV1Request instanceGroupV1Request = new InstanceGroupV1Request();
+        instanceGroupV1Request.setRecipeNames(null);
+        distroXV1Request.setInstanceGroups(Collections.singleton(instanceGroupV1Request));
+        distroXV1Request.getAllRecipes();
```
Comment (keyki) [inline core-api/src/test/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXV1RequestTest.java:17]: Shouldn't we verify that the result is an empty set?

Comment (sodre90) [reply core-api/src/test/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXV1RequestTest.java]: We can verify, I just wanted to create as minimal unit test as I can to test this use case :D anyway, assert is added

**Review by keyki — APPROVED** [review]

**Pull request merged**